modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/nativelib/ArtifactCopyNativeLibraryFinder.java

```diff
@@ -124,8 +124,8 @@ private String findLibraryLocally(String name) {
           } catch (URISyntaxException e) {
             return false;
           }
-          String baseName = getBaseName(fullPath);
-          return baseName.contains(name) && (fullPath.endsWith(extension) || IS_OS_MAC && fullPath.endsWith(JNILIB_EXTENSION));
+          String baseName = getLibraryBaseName(fullPath);
```

Comment:
Let's try to maintain the "loose" matching in case we do not find a "strict" one. I'm afraid we can break old apps otherwise.